sw/device/silicon_creator/manuf/lib/individualize.c

```diff
@@ -237,21 +237,21 @@ status_t manuf_individualize_device_secret0(
     return OK_STATUS();
   }
 
-  uint64_t hashed_test_unlock_token[kSecret0TestUnlockTokenSizeIn64BitWords];
-  uint64_t hashed_test_exit_token[kSecret0TestExitTokenSizeIn64BitWords];
-  TRY(manuf_util_hash_lc_transition_token(provisioning_data->test_unlock_token,
-                                          kSecret0TestUnlockTokenSizeInBytes,
-                                          hashed_test_unlock_token));
+  uint32_t hashed_test_unlock_token[kSecret0TestUnlockTokenSizeIn32BitWords];
+  uint32_t hashed_test_exit_token[kSecret0TestExitTokenSizeIn32BitWords];
+  TRY(manuf_util_hash_lc_transition_token(
+      provisioning_data->test_unlock_token,
+      kSecret0TestUnlockTokenSizeIn32BitWords, hashed_test_unlock_token));
   TRY(manuf_util_hash_lc_transition_token(provisioning_data->test_exit_token,
-                                          kSecret0TestExitTokenSizeInBytes,
+                                          kSecret0TestExitTokenSizeIn32BitWords,
                                           hashed_test_exit_token));
 
-  TRY(otp_ctrl_testutils_dai_write64(
+  TRY(otp_ctrl_testutils_dai_write32(
```

Comment:
I think because the `secret*` partitions have 64-bit access granules, this will fail checks in the testutils/DIF SW layer. (Same with other similar writes). Can we perhaps provide a 64-bit word buffer interface for some crypto operations? (like what you have done for the 32-bit) Or maybe you have another suggestion?

Reply:
Well, we could add a flag to disable strict aliasing and it wouldn't be an issue (see https://github.com/lowRISC/opentitan/issues/16556) -- I could keep the 64-bit buffers if it didn't violate strict aliasing to cast `uint64_t *` to `uint32_t *`. Morally, all I need is a 32-bit aligned buffer, and a 64-bit aligned buffer _is_ 32-bit aligned.

I'll put the rest of my thoughts on that issue to keep discussion in the same place, but I'm coming to the opinion that we should really add the flag and just disable strict aliasing.